hooks/pre/05-check_scenario_requirements.rb

```diff
@@ -0,0 +1,84 @@
+require 'json'
+require 'pathname'
+require 'kafo/puppet_command'
+
+PREFIX = %W(PiB TiB GiB MiB KiB B).freeze
+
+# From https://codereview.stackexchange.com/questions/9107/printing-human-readable-number-of-bytes
+def as_size(s)
+  s = s.to_f
+  i = PREFIX.length - 1
+  while s > 512 && i > 0
+    i -= 1
+    s /= 1024
+  end
+  ((s > 9 || s.modulo(1) < 0.1 ? '%d' : '%.1f') % s) + ' ' + PREFIX[i]
+end
+
+# Based on https://stackoverflow.com/a/26878510
+def is_descendant?(a, b)
+  return true if a == '/'
+
+  a_list = File.expand_path(a).split('/')
+  b_list = File.expand_path(b).split('/')
+
+  b_list[0..a_list.size-1] == a_list
+end
+
+def facts(key)
+  command = Kafo::PuppetCommand.search_puppet_path('facter')
+  raise Exception, 'Facter not found' unless command
+  JSON.load(`#{command} --json error #{key}`)[key]
+end
+
+def check
+  # TODO: read required values from the scenario
+  #scenario = kafo.class.scenario_manager.select_scenario
+
+  required = {}
+  #required = {
+  #  'memory' => 8 * 1024 * 1024 * 1024,
```

Comment:
FYI, you want to check for less than 8G, more something like 7.8 or 7.9G, as if people have 8G memory, some of it might be claimed by kdump/crashkernel.

Reply:
That's why below I do a ```required * 0.9```